### PR TITLE
feat: use solady proxies

### PR DESCRIPTION
**Pull request by adamegyed**
## Motivation

Optimizing code size and runtime gas.

## Solution

Switch UUPSUpgradeable from the OZ implementation to the solady implementation. This removes the public function `UPGRADE_INTERFACE_VERSION`, but keeps the implementations of `proxiableUUID` and `upgradeToAndCall`.

For ModularAccount specifically, switch from using OZ `ERC1967` proxies to using solady's LibClone to create ERC1967 proxies. This also reduces codesize on the factory, and prevents us from having to compute the proxy bytecode hash in the constructor. SemiModularAccount was already using this library, as it was necessary for appending immutable args.

### CODE REVIEW

File: gas/modular-account/ModularAccount.gas.t.sol [hunk 1]
```diff
@@ -28,19 +28,18 @@ contract ModularAccountGasTest is ModularAccountBenchmarkBase("ModularAccount")
         // Also assert that the event emitted by the factory is correct
         Vm.Log[] memory logs = vm.getRecordedLogs();
 
-        assertEq(logs.length, 5);
+        assertEq(logs.length, 4);
         // Logs:
-        // 0: ERC1967Proxy `Upgraded`
```
Comment (adamegyed) [inline gas/modular-account/ModularAccount.gas.t.sol:33]: For reviewers: The soladay proxy doesn't emit this on creation. The solady UUPSUpgradeable emits it as expected on any subsequent upgrades.

File: src/factory/AccountFactory.sol [hunk 2]
```diff
@@ -47,24 +42,25 @@ contract AccountFactory is Ownable {
      */
     function createAccount(address owner, uint256 salt, uint32 entityId) external returns (ModularAccount) {
```
Comment (Zer0dot) [inline src/factory/AccountFactory.sol:43]: unrelated, but what was the point of including the entityId here again?

**Review by Zer0dot — APPROVED** [review]

File: gas/modular-account/ModularAccount.gas.t.sol (hunk 1 above)
Comment (Zer0dot) [reply gas/modular-account/ModularAccount.gas.t.sol:33]: Shouldn't be a problem right

File: src/factory/AccountFactory.sol (hunk 2 above)
Comment (adamegyed) [reply src/factory/AccountFactory.sol:43]: Not sure tbh... we could probably either remove it, or offer two functions and have one of them pick a default (likely 0).

File: gas/modular-account/ModularAccount.gas.t.sol (hunk 1 above)
Comment (adamegyed) [reply gas/modular-account/ModularAccount.gas.t.sol:33]: I think it's ok, we don't use this in any of our offchain services.

**Pull request merged**